### DISCUSSION TITLE
CVE-2020-13902

### DISCUSSION THREAD

**Discussion by pgajdos** [ImageMagick/ImageMagick#2132] [General]
This CVE was assigned to
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=20920

Could I ask you for your statement?

(Nope, I cannot reproduce anyhow with the testcase even with 7.0.10-17.)

**Comment by dlemstra** [comment]
This appears to be happening because the tiff library changed the data type of `TIFFTAG_RICHTIFFIPTC`. The regression range reported by google fuzz is this:

https://gitlab.com/libtiff/libtiff/compare/b4820b2179641e8638a9aac8105a4e1ea50ecbfa...06d6e36187f465f8fd2b2140aefc13e79c3cfb89. 

As you can see they changed the type from `TIFF_SETGET_C32_UINT32` to `TIFF_SETGET_C32_UINT8`. This was done due to this issue: https://gitlab.com/libtiff/libtiff/-/issues/78 and changed in this pr: https://gitlab.com/libtiff/libtiff/-/merge_requests/99. But that breaks our code because we assume that we need to multiply the size that we receive by 4. This patch should fix the issue: https://github.com/ImageMagick/ImageMagick/commit/824f344ceb823e156ad6e85314d79c087933c2a0 and will be available in the next version (7.0.10-20).
